src/sna/sna_display.c

```diff
@@ -3108,7 +3108,7 @@ prefer_non_hidpi(xf86OutputPtr output, DisplayModePtr Modes)
 		return;
 
 	/* Check if hiDPI */
-	if (preferred->HDisplay != 3840 || preferred->VDisplay != 2160)
+	if (preferred->HDisplay < 3200 || preferred->VDisplay < 1800)
```

Comment:
I'm wondering if we shouldn't actually check for `(preferred->HDisplay <= 1920 || preferred->VDisplay < 1080)` and cover all HiDPI resolutions already (WQHD aka 2k is 2560x1440).